### PR TITLE
Chore: Convert AppSetting to tsx

### PR DESCRIPTION
**Pull request by juliajforesti**

### CODE REVIEW

File: apps/meteor/client/views/admin/apps/AppSetting.tsx [hunk 1]
```diff
@@ -1,41 +1,64 @@
...
 
 import MarkdownText from '../../../components/MarkdownText';
 import MemoizedSetting from '../settings/MemoizedSetting';
 
-const useAppTranslation = (appId) => {
+type AppTranslationFunction = {
+	(key: string, ...replaces: unknown[]): string;
+	has: (key: string | undefined) => boolean;
+};
+const useAppTranslation = (appId: string): AppTranslationFunction => {
```
Comment (tassoevan) [inline apps/meteor/client/views/admin/apps/AppSetting.tsx:12]: `ReturnType<typeof useTranslation>` might be ok too, I guess...

File: apps/meteor/client/views/admin/apps/AppSetting.tsx [hunk 2]
```diff
@@ -1,41 +1,64 @@
...
+type AppSettingProps = {
+	appSetting: {
+		id: string;
+		type: ISettingBase['type'];
+		i18nLabel: string;
+		i18nDescription: string;
+		values: { key: string; i18nLabel: string }[];
+		required: boolean;
+	};
+	onChange: any;
```
Comment (tassoevan) [inline apps/meteor/client/views/admin/apps/AppSetting.tsx:56]: image

File: apps/meteor/client/views/admin/apps/AppSetting.tsx [hunk 3]
```diff
@@ -66,7 +89,7 @@ function AppSetting({ appSetting, onChange, value, ...props }) {
 			value={value}
 			onChangeValue={onChange}
 			_id={id}
-			values={translatedValues}
+			{...({ values: translatedValues } as any)}
```
Comment (tassoevan) [inline apps/meteor/client/views/admin/apps/AppSetting.tsx:92]: > any

image

File: apps/meteor/client/views/admin/apps/AppSetting.tsx (hunk 1 above)
Comment (juliajforesti) [reply apps/meteor/client/views/admin/apps/AppSetting.tsx:12]: almost... typeof has are incompatible :(

File: apps/meteor/client/views/admin/apps/AppSetting.tsx (hunk 2 above)
Comment (juliajforesti) [reply apps/meteor/client/views/admin/apps/AppSetting.tsx:56]: ooops, forgot to type it :)

File: apps/meteor/client/views/admin/apps/AppSetting.tsx (hunk 3 above)
Comment (juliajforesti) [reply apps/meteor/client/views/admin/apps/AppSetting.tsx:92]: That was acctually intentional... but fixed it anyway.

**Review by rique223 — APPROVED** [review]
LGTM!

**Review by ggazzo — APPROVED** [review]

**Pull request merged**